doc/server.md

```diff
@@ -0,0 +1,124 @@
+| :--           | :--         | :--
+| `-H, --host HOST` | `localhost` | The hostname of the webserver endpoint
+| `-P, --port PORT` | `8080`      | The port number of the webserver endpoint; must be between 0 and 65536
+| `-h, --help`      | N/A         | Display the top-level help guide
+
+The `validate` subcommand starts the server in validate mode, where any Statements sent to the `/statements` endpoint will undergo Template matching against the Profiles that the server was given on startup. If a Statement array is sent, only the last statement will be validated. The following table shows the arguments to `validate`:
+
+| Subcommand&nbsp;Argument | Description
+| :--                     | :--
+| `-p, --profile URI`     | Profile URI filepath/location; must specify one or more.
+| `-i, --template-id IRI` | IDs of Statement Templates to validate against; can specify zero or more. Filters out all Templates that are not included.
+| `-a, --all-valid`       | If set, any Statement is not considered valid unless it is valid against ALL Templates. Otherwise, a Statement only needs to be valid against at least one Template.
+| `-c, --short-circuit`   | If set, then print on only the first Template any Statement fails validation against.Otherwise, print for all Templates a Statement fails against.
+| `-h, --help`            | Display the 'validate' subcommand help menu.
+
+The `match` subcommand starts the server in match mode, where any Statements sent to the `/statements` endpoint will undergo Pattern matching against the Profiles that the server was given on startup. If a single Statement is sent, it is coerced into a Statement batch. The following table shows the arguments to `match`:
+
+| Subcommand&nbsp;Argument | Description
+| :--                    | :--
+| `-p, --profile URI`    | Profile filepath/location; must specify one or more.
+| `-i, --pattern-id IRI` | IDs of primary Patterns to match against; can specify zero or more. Filters out all Patterns that are not included.
+| `-h, --help`           | Display the 'match' subcommand help menu.
+
+In addition to the `POST /statements` endpoint, there is a `GET /health` endpoint that is used to perform a server health check:
+```
+% curl http://0.0.0.0:8080/health
+```
+which will return an response with status `200` and body `OK`.
+
+There is no `PUT` or `GET` versions of the `/statements` endpoint, unlike what is required in a learning record store.
+
+# Examples for validate mode
+
+For the first few examples, let us start a webserver in validate mode with a single Profile. Assume that we have already copied the contents of `test-profile` into the `target/bundle` directory. Running this command
+```
+% ./bin/server.sh validate --profile sample_profiles/calibration.jsonld
+```
+will start up a server in validate mode on `localhost:8080` with a single Profile set to validate against.
+
+To validate a single Statement against Templates in that Profile:
+```
+% curl localhost:8080/statements \
```

Comment:
I would suggest adding the `-i` flag to the cURL calls here so you get the whole response + code like:

```http
HTTP/1.1 204 No Content
Date: Mon, 01 May 2023 15:57:03 GMT
Strict-Transport-Security: max-age=31536000; includeSubdomains
X-Frame-Options: DENY
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
X-Download-Options: noopen
X-Permitted-Cross-Domain-Policies: none
Content-Security-Policy: object-src 'none'; script-src 'unsafe-inline' 'unsafe-eval' 'strict-dynamic' https: http:;
```

Reply:
Eh, most of that stuff seems like extraneous information.

Reply:
It's more like, the docs say it's gonna be a 204, you should show that it is indeed a 204 rather than the default behavior of just showing the body, which is in this case nothing

Reply:
Ok I could add to the first example "if you set the `-i` flag for `curl` you'll get this additional information." (I don't think we need to do so for _every_ example.)

Reply:
Though wouldn't it vary by user?

Reply:
> Ok I could add to the first example "if you set the `-i` flag for `curl` you'll get this additional information." (I don't think we need to do so for _every_ example.)

That's too much (the explanation). I was thinking more about just adding the arg to the example to make it clearer to someone going through the doc, but in the grand scheme of things it doesn't matter either way. If it bothers you, ignore this comment.

Reply:
I added response headers to the webserver docs; let me know if it's too much: https://github.com/yetanalytics/project-persephone/pull/40/commits/1757eef05a7d0786fdf97168fbfa5018f509a064